pkg/gcp/client/compute.go

```diff
@@ -538,3 +553,41 @@ func (c *computeClient) ListImages(ctx context.Context, imageName, orderBy, fiel
 func (c *computeClient) GetRegion(ctx context.Context, region string) (*compute.Region, error) {
 	return c.service.Regions.Get(c.projectID, region).Context(ctx).Do()
 }
+
+// WaitForIPv6Cidr waits for the ipv6 cidr block association
+func (c *computeClient) WaitForIPv6Cidr(ctx context.Context, region, subnetID string) (string, error) {
+	maxRetries := 30
+	waitInterval := 10 * time.Second
+	for i := 0; i < maxRetries; i++ {
+		select {
+		case <-ctx.Done():
+			return "", ctx.Err()
+		case <-time.After(waitInterval):
+			ipv6CidrBlock, err := c.GetIPv6CidrForSubnet(ctx, region, subnetID)
+			if err == nil {
+				return ipv6CidrBlock, nil
+			}
+			if !IsRetryableIPv6CIDRError(err) {
+				return "", err
+			}
+		}
+	}
```

Comment:
Nit: I believe the k8s packages already offer a a wait.RetryUntil type of function.